woocommerce/woocommerce-ios
# [Mobile Payments] Bump WCPay minimum from 2.5 to 2.9 for test_mode for WCPayAccount

Partially addresses #4428 

@jaclync @joshheald - only one review needed - both of y'all are welcome of course cc @ctarda 

This is the first in a series of PRs to notify merchants that in-person payments are not supported on live accounts that are set to test mode.

Changes in this PR:

- This PR decodes the is_live and test_mode flags in the `wc/v3/payments/accounts` response ([test_mode](https://github.com/Automattic/woocommerce-payments/blame/6eaca4f05faa07acd0392a32ac1b1af051f93ca3/includes/admin/class-wc-rest-payments-accounts-controller.php#L73) was added in WCPay 2.9 and is_live is provided by our WPCOM backend - see compose_basic_account_info in the server repo)
- Updates unit tests and sample jsons to wit
- Per https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html "When strings are used for raw values, the implicit value for each case is the text of that case’s name." - so I removed the unneeded strings for `country` and `supported` in the coding keys for `WCPayAccount`

Changes coming in later PRs:

The next PR will add these Bools to PaymentGatewayAccount. This will also require adding them as attributes to that Core Data entity.

A third PR will finish implementing isWCPayInTestModeWithLiveStripeAccount in CardPresentPaymentsOnboardingUseCase and a SwiftUI view to prompt the user that “In-Person Payments is Not Available When Test Mode is Enabled on Your Site” or somesuch, similar to the other “InPersonPayments*” views in Onboarding Errors

To test:
- Nothing user facing is changing with this PR. To test, ensure unit tests pass

Update release notes:
